client/app/students/students.html

```diff
@@ -84,7 +84,12 @@
                     <td md-cell ng-if="isEditor()" ng-click="edit($event, item, 'avgGrade')">{{ item.avgGrade }}</td>
                     <td md-cell ng-if="!isEditor()">{{ item.avgGrade }}</td>
                     <td md-cell ng-if="!isEditor()">{{ item.social }}</td>
-                    <td md-cell ng-if="isEditor()" ng-click="edit($event, item, 'social')" >{{ item.social }} </td>
+                    <td md-cell ng-if="isEditor()">
+                        <md-input-container>
```

Comment:
You should use edit inside the cell rather than put a simple input.
See that you can add your constraints like min max in the controller inside the $scope.validators (I see that you did, you should use it with the edit function)